src/inventory/managers/InventoryWindowManager.ts

```diff
@@ -0,0 +1,77 @@
+import GameUI from "@/scenes/GameUI";
+import InventoryGridManager from "./InventoryGridManager";
+import { GameEntity } from "@/GameEntity";
+import { InventoryGridSlotEvent } from "../events/InventoryGridSlotEventEmitter";
+import { InventoryEvent } from "../events/InventoryEventsEmitter";
+import { inventoryEvents, moveItemToSlot } from "../state/InventoryUtilities";
+import { decomposeItem, getValidDropTarget } from "../Utils";
+
+export default class InventoryWindowManager {
+  constructor(
+    private readonly scene: GameUI,
+    private readonly playerInventory: InventoryGridManager
+  ) {
+    inventoryEvents.on(InventoryEvent.ITEM_ADDED, (item: GameEntity) => {
+      this.playerInventory.addItem(decomposeItem(item));
+    });
+
+    inventoryEvents.on(InventoryEvent.ITEM_REMOVED, (itemMovedProps) => {
+      this.playerInventory.removeItem(itemMovedProps.currentSlotIndex);
+    });
+
+    inventoryEvents.on(InventoryEvent.HIDE_ITEM, (hideItemProps) => {
+      if (hideItemProps.hide) {
+        this.playerInventory.hideItem(hideItemProps.currentSlotIndex);
+      } else {
+        this.playerInventory.showItem(hideItemProps.currentSlotIndex);
+      }
+    })
+
+    // playerInventory.slots.forEach((slot) => {
```

Comment:
This will be used later?

Reply:
If we decide to implement the drag and drop it will